### ISSUE TITLE
skaffold doesn't check for helm release presence in same namespace that it deploys to.

### ISSUE THREAD

**Issue by clement-buchart** [GoogleContainerTools/skaffold#3838]
### Expected behavior

`skaffold deploy` to work even though helm releases already exists.

### Actual behavior

GOTO https://github.com/GoogleContainerTools/skaffold/issues/3838#issuecomment-600621901

We're having an issue while upgrade our helm releases with skaffold, which cause these logs : 
```
Starting deploy...
Helm release postgresql not installed. Installing...
Error: cannot re-use a name that is still in use
time="2020-03-18T11:51:08+01:00" level=warning msg="exit status 1" 
```
Even though the helm release does exist.

After some troubleshooting, it appeared that skaffold incorrectly think that the helm release doesn't exists and try to install it instead of upgrading it, I suppose because of the following piece of code that expect any error during the `helm get` to be caused by the non-existence of the helm release ; 
https://github.com/GoogleContainerTools/skaffold/blob/95cfed1d047bed0a578cb84a414be071333e9884/pkg/skaffold/deploy/helm.go#L260-L262

When setting log level to debug, it appears that the helm command built to check for release existence isn't valid : 
`DEBU[0002] Running command: [helm --kube-context  get all postgresql --kubeconfig /tmp/totokube]`
The empty kubecontext causing 
```
helm.go:71: context "get" does not exist
```
It appears the --kube-context flag is passed to helm even though it is empty, which is a different behaviour from namespace and kubeconfig flags.
https://github.com/GoogleContainerTools/skaffold/blob/95cfed1d047bed0a578cb84a414be071333e9884/pkg/skaffold/deploy/helm.go#L223-L230

I suspect that this cause my problem, since the helm command then fails, leading the the release being considered new.

How would you feel about checking for --kubecontext emptiness before passing it to helm ? Is there a reason for the current behaviour that I'm missing ?

I suppose another fix would be to explicitly set kubeContext the the default kubeContext of the kubeconfig file provided, if any, here : 
https://github.com/GoogleContainerTools/skaffold/blob/6a92475a4aa07180b1340c5bb1aa2b18ae5058ca/pkg/skaffold/kubernetes/context/context.go#L47-L62

Any insights appreciated.
Thanks

### Information

- Skaffold version: 1.5.0
- Operating system: Arch Linux
- Contents of skaffold.yaml: N/A

**Comment by clement-buchart** [comment]
Nevermind, the issue is actually that skaffold test for release existence in the namespace specified in skaffold.yaml, and not in the one passed via CLI : 
https://github.com/GoogleContainerTools/skaffold/blob/95cfed1d047bed0a578cb84a414be071333e9884/pkg/skaffold/deploy/helm.go#L260

This seems counter-intuitive given what's specified in the [documentation](https://skaffold.dev/docs/references/cli/) : 
`  -n, --namespace='': Run deployments in the specified namespace`

What do you think ?

**Comment by nkubala** [comment]
closed through #3914
